src/hyperlink/_url.py

```diff
@@ -29,16 +29,15 @@
 except ImportError:  # Python 2
     from collections import Mapping
 
-# Note: IDNAError is a subclass of UnicodeError
-from idna import encode as idna_encode, decode as idna_decode, IDNAError
```

Comment:
IDNAError was unused

Reply:
Yeah I think at one point there may have been thoughts of catching and reraising, but now we just let them through.